float32 width
float32 speed
float32 force
float32 epsilon_inner
float32 epsilon_outer
---
bool success